### PR TITLE
modify migrate-reporting for changes to configurable subjects

### PR DESCRIPTION
**Pull request by malaffoon**
Most of the changes are to tests and setting up the data for them.

### CODE REVIEW

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagingToReportingStepsConfig.java [hunk 1]
```diff
@@ -59,7 +59,6 @@ public Step deleteEntitiesStep() {
                 "item",
                 "item_other_target",
                 "item_common_core_standard",
-                "asmt_score",
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagingToReportingStepsConfig.java]: as noted in the schema changes, this table is 100% redundant with the denormalized columns in `asmt`

File: migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml [hunk 2]
```diff
@@ -631,50 +652,80 @@ sql:
...
+                LEFT JOIN (SELECT s.exam_id, s.scale_score, s.scale_score_std_err, s.performance_level
+                             FROM staging_exam_score s JOIN subject_score m ON m.id = s.subject_score_id AND m.score_type_id=3 AND m.data_order=5
                           ) AS claim5 ON claim5.exam_id = se.id
-                LEFT JOIN (SELECT s.exam_id, s.scale_score, s.scale_score_std_err, s.category
-                             FROM staging_exam_claim_score s
-                               INNER JOIN subject_claim_score m ON m.id = s.subject_claim_score_id AND m.data_order = 6
+                LEFT JOIN (SELECT s.exam_id, s.scale_score, s.scale_score_std_err, s.performance_level
+                             FROM staging_exam_score s JOIN subject_score m ON m.id = s.subject_score_id AND m.score_type_id=3 AND m.data_order=6
                           ) AS claim6 ON claim6.exam_id = se.id
+                LEFT JOIN (SELECT s.exam_id, s.scale_score, s.scale_score_std_err, s.performance_level
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml]: This is really the part that concerns me about the exam table being wide and denormalized: 12 subselects. Of course, this is only being done on the data being migrated, and each subselect produces 0-or-1 rows.

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 3]
```diff
@@ -211,88 +212,77 @@ sql:
         # ---------------- subject_asmt_type  -----------------------------------------------
         subject_asmt_type:
           sql:
-            # TODO - deal with subject_asmt_scoring and get rid of joins here
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: TODOne!

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/UpsertSubjectsIT.java [hunk 4]
```diff
@@ -12,6 +12,7 @@
 import org.springframework.test.context.jdbc.SqlGroup;
 
 import java.util.List;
+import java.util.Map;
```
Comment (malaffoon) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/UpsertSubjectsIT.java]: oops, not sure why this snuck in, will remove

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagePackageIT.java [hunk 5]
```diff
@@ -17,6 +17,7 @@
 
 import java.time.Instant;
 import java.util.List;
+import java.util.Map;
```
Comment (malaffoon) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/StagePackageIT.java]: will remove

File: migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml (hunk 2 above)
Comment (esotrope) [reply migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml]: why can't we just join in this case?

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml (hunk 3 above)
Comment (esotrope) [reply migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: woo!

File: migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml (hunk 2 above)
Comment (malaffoon) [reply migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml]: They might not exist. Alt scoring is optional.

**Pull request merged**